### ISSUE TITLE
False-positive SinglePipe when piped after `for` special form

### ISSUE THREAD

**Issue by eksperimental** [rrrene/credo#1163]
### Environment

* Credo version (`mix credo -v`): 1.7.10-ref.main.b1010b5+uncommittedchanges
* Erlang/Elixir version (`elixir -v`): Elixir 1.17.3 (compiled with Erlang/OTP 27)
* Operating system:

### What were you trying to do?

`mix credo --strict`

### Expected outcome

```elixir
for {major, patches} <- versions do
  # ...
end
|> Enum.reverse()
```

### Actual outcome

```
  Code Readability
┃
┃ [R] ↗ Use a function call when a pipeline is only one function long.
```

Even with `{Credo.Check.Readability.SinglePipe, [allow_0_arity_functions: true]}` config this does not show the issue.

**Comment by rrrene** [comment]
This is now available in Credo `1.7.15`! 😄 🎉
